### ISSUE TITLE
Consolidate MacOS trunk jobs into fewer and bigger jobs

### ISSUE THREAD

**Issue by huydhn** [pytorch/executorch#2898]
Atm, MacOS trunk jobs are spawned individually for each models.  This is a simple setup, but it has a "big" drawback that there are many small jobs (cc @DanilBaibak to supply the data), which is not ideal because:

1. All available MacOS runners are quickly used up by the large number of jobs.
2. They waste time checking out ET and PT, then run only for a few minutes.

We should work on grouping these individual jobs into fewer and bigger ones, maybe an hour each.  Each job could check more than one models before returning.

cc @guangy10

**Comment by guangy10** [comment]
Yes, there is a plan to revisit ExecuTorch CI setup after Alpha release.

**Comment by digantdesai** [comment]
@huydhn @guangy10 I am assuming this is stale, and we can close this? Feel free to open is this is not the case. Thanks.
